### PR TITLE
Process Data in Models

### PR DESCRIPTION
**Pull request by sharon-odhiambo**
- In this PR I created models, inserted data and use CRUD and other custom methods to process data in the models.

**Below are the highlights:**

- Create models for users, comments, likes and posts
- Create associations between models
- Insert data into the models
- Use CRUD method s to process the data
- Create custom methods for counting posts in users, and comments and likes in posts

Cheers and Happy reviewing!

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  has_many :comments
+  has_many :likes
+  belongs_to :users, class_name: 'User'
+
```
Comment (zainsadaqat) [inline app/models/post.rb:6]: ### [OPTIONAL]
You can use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_post_counter method every time a new post has been saved to the database, then make the update_post_counter method private
```ruby
  # app/models/post.rb
  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end
```

File: app/models/comment.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class Comment < ApplicationRecord
+  belongs_to :users, class_name: 'User'
+  belongs_to :posts
+
```
Comment (zainsadaqat) [inline app/models/comment.rb:5]: ### [OPTIONAL]
**Hint** : Just like we have in posts, we can do the same here `after_save :update_comments_counter`

File: app/models/like.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class Like < ApplicationRecord
+  belongs_to :users
+  belongs_to :posts
+
```
Comment (zainsadaqat) [inline app/models/like.rb:5]: ### [OPTIONAL]
**Hint** : Just like we have in posts, we can do the same here `after_save :update_likes_counter`

**Review by zainsadaqat — APPROVED** [review]
# APPROVED!!!! 🎉 🎉 🎉 🎉 🎉 🎉
## Hey @sharon-odhiambo  😊 😃 

#### Great Job 😄 👏 You've done it. Your project is complete!  There is nothing else to say other than... it's time to merge it 👍🏻 ✅ 

# Congratulations!!! 🎉 🥳 👏 🙌 🎊 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by sharon-odhiambo** [comment]
Hi @zainsadaqat , thank you for the review and the additional suggestions. Let me check them out.

**Pull request merged**